src/parser.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+import requests as rq
+import json
+from bs4 import BeautifulSoup as bs
+import configuration as cf
+from transformers import pipeline
+
+models: dict = {}
+
+
+def load_models():
+    question_name = "deepset/roberta-base-squad2"
+    models["qa"] = pipeline("question-answering", model=question_name, tokenizer=question_name)
+    translation_name = "Helsinki-NLP/opus-mt-ru-en"
+    models["trans"] = pipeline("translation", model=translation_name, tokenizer=translation_name)
+    classifier_name = "joeddav/xlm-roberta-large-xnli"
+    models["class"] = pipeline("zero-shot-classification", model=classifier_name)
+
+
+def pages_num(soup: bs) -> int:
+    pagination = soup.find("div", class_="PaginationWidget__wrapper-pagination")
+    paginator = pagination.find_all("a")
+    num = 0
+    for data_page in paginator:
+        if num < int(data_page["data-page"]):
+            num = int(data_page["data-page"])
+    return num
+
+
+def parse(to_file: str):
+    try:
+        f = open(to_file, "w")
+    except IOError as e:
+        return
+
+    if not models:
+        load_models()
+
+    data: dict[str, dict[str, dict]] = {}
+
+    candidate_labels = [
+        "Модуль памяти",
+        "Видеокарта",
+        "Процессор",
+        "Материнская плата",
+        "Блок питания",
+        "Корпус",
+    ]
+
+    url_template = "https://www.citilink.ru/catalog/"
+    matrix = {
+        cf.MB: "materinskie-platy/",
+        cf.CPU: "processory/",
+        cf.GPU: "videokarty/",
+        cf.RAM: "moduli-pamyati/",
+        cf.PB: "bloki-pitaniya/",
+        cf.BODY: "korpusa/",
+    }
```

Comment:
In the case of class, these could be private attributes => function will be more accurate and clean

Reply:
Yes!